### PR TITLE
feat(TBD-8427):CDH 6.1.1 Spark

### PR DESCRIPTION
**Pull request by lbourgeois**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**BREAKING CHANGE**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/cdh/IClouderaDistribution.java [hunk 1]
```diff
@@ -12,6 +12,7 @@
 // ============================================================================
 package org.talend.hadoop.distribution.constants.cdh;
 
+
```
Comment (CarlosTalend) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/cdh/IClouderaDistribution.java]: Was there anything to be added here?

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/template/modulegroup/node/sparkstreaming/DynamicSparkStreamingKafkaAssemblyModuleGroup.java [hunk 2]
```diff
@@ -46,9 +46,16 @@ protected void init() {
...
                 SparkStreamingConstant.KAFKA_SPARKCONFIGURATION_LINKEDPARAMETER, "SUPPORTED_SPARK_VERSION", EqualityOperator.EQ, //$NON-NLS-1$
                 ESparkVersion.SPARK_1_6.getSparkVersion()));
-        spark2Condition = new SimpleComponentCondition(new LinkedNodeExpression(
-                SparkStreamingConstant.KAFKA_SPARKCONFIGURATION_LINKEDPARAMETER, "SUPPORTED_SPARK_VERSION", EqualityOperator.EQ, //$NON-NLS-1$
-                ESparkVersion.SPARK_2_2.getSparkVersion()));
+        
+        spark2Condition = new MultiComponentCondition(
+        		new LinkedNodeExpression(SparkStreamingConstant.KAFKA_SPARKCONFIGURATION_LINKEDPARAMETER,
+                        "SUPPORTED_SPARK_VERSION", EqualityOperator.EQ, //$NON-NLS-1$
+                        ESparkVersion.SPARK_2_2.getSparkVersion()), //$NON-NLS-1$
```
Comment (CarlosTalend) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/template/modulegroup/node/sparkstreaming/DynamicSparkStreamingKafkaAssemblyModuleGroup.java]: Is it normal that Spark 2.3 isn't present here?

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/template/modulegroup/node/sparkstreaming/DynamicSparkStreamingKafkaClientModuleGroup.java [hunk 3]
```diff
@@ -46,9 +46,16 @@ protected void init() {
...
                 ESparkVersion.SPARK_1_6.getSparkVersion()));
-        spark2Condition = new SimpleComponentCondition(new LinkedNodeExpression(
-                SparkStreamingConstant.KAFKA_SPARKCONFIGURATION_LINKEDPARAMETER, "SUPPORTED_SPARK_VERSION", EqualityOperator.EQ, //$NON-NLS-1$
-                ESparkVersion.SPARK_2_2.getSparkVersion()));
+        
+        spark2Condition = new MultiComponentCondition(
+        		new LinkedNodeExpression(SparkStreamingConstant.KAFKA_SPARKCONFIGURATION_LINKEDPARAMETER,
+                        "SUPPORTED_SPARK_VERSION", EqualityOperator.EQ, //$NON-NLS-1$
+                        ESparkVersion.SPARK_2_2.getSparkVersion()), //$NON-NLS-1$
+                BooleanOperator.OR,
```
Comment (CarlosTalend) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/template/modulegroup/node/sparkstreaming/DynamicSparkStreamingKafkaClientModuleGroup.java]: Ditto pour Spark 2.3

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/template/modulegroup/node/sparkstreaming/DynamicSparkStreamingKinesisNodeModuleGroup.java [hunk 4]
```diff
@@ -34,17 +36,23 @@
...
         init();
     }
 
     protected void init() {
-        spark2Condition = new SimpleComponentCondition(
+        spark2Condition = new MultiComponentCondition(
+        		new LinkedNodeExpression(SparkStreamingConstant.SPARK_STREAMING_SPARKCONFIGURATION_LINKEDPARAMETER,
+                        "SUPPORTED_SPARK_VERSION", EqualityOperator.EQ, //$NON-NLS-1$
+                        ESparkVersion.SPARK_2_2.getSparkVersion()), //$NON-NLS-1$
+                BooleanOperator.OR,
```
Comment (CarlosTalend) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/template/modulegroup/node/sparkstreaming/DynamicSparkStreamingKinesisNodeModuleGroup.java]: Ditto pour Spark 2.3

File: main/plugins/org.talend.hadoop.distribution.cdh5x/resources/builtin/cdhx/Cloudera_CDH6_1_1.json [hunk 5]
```diff
@@ -3,32 +3,33 @@
     "childNodes" : [ ],
     "distribution" : "CLOUDERA",
     "id" : "Cloudera_CDH6_1_1",
-    "name" : "Cloudera CDH6.1.1 (Dynamic)",
-    "projectTechnicalName" : "LOCAL_PROJECT",
+    "name" : "Cloudera CDH6.1.1",
     "repository" : "https://talend-update.talend.com/nexus/content/groups/dynamicdistribution/",
```
Comment (CarlosTalend) [inline main/plugins/org.talend.hadoop.distribution.cdh5x/resources/builtin/cdhx/Cloudera_CDH6_1_1.json]: Is this one enough, or should there be another repository like https://talend-update.talend.com/nexus/content/repositories/cdh-releases-rcs/ , used in 5.13?

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/template/modulegroup/node/sparkstreaming/DynamicSparkStreamingKafkaAssemblyModuleGroup.java (hunk 2 above)
Comment (aparent-talend) [reply main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/template/modulegroup/node/sparkstreaming/DynamicSparkStreamingKafkaAssemblyModuleGroup.java]: Yes as it is not officialy supported by any built-in in the dynamic distribution

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/template/modulegroup/node/sparkstreaming/DynamicSparkStreamingKafkaClientModuleGroup.java (hunk 3 above)
Comment (aparent-talend) [reply main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/template/modulegroup/node/sparkstreaming/DynamicSparkStreamingKafkaClientModuleGroup.java]: same answer than before

File: main/plugins/org.talend.hadoop.distribution.cdh5x/resources/builtin/cdhx/Cloudera_CDH6_1_1.json (hunk 5 above)
Comment (aparent-talend) [reply main/plugins/org.talend.hadoop.distribution.cdh5x/resources/builtin/cdhx/Cloudera_CDH6_1_1.json]: this one is enough, this is a proxy repository for both cloudera and hortonworks repos

**Pull request merged**